NeoML/include/NeoML/Dnn/Layers/FullyConnectedLayer.h

```diff
@@ -77,8 +79,16 @@ class NEOML_API CFullyConnectedLayer : public CBaseLayer {
 	CPtr<CDnnBlob>& FreeTermsDiff() { return paramDiffBlobs[1]; }
 
 private:
-	int numberOfElements; // the number of elements (neurons) of the fully-connected layer
-	bool isZeroFreeTerm; // indicates if the free term should be set to zero
+	enum TSMMD { TSMMD_Forward, TSMMD_Backward, TSMMD_Learn, /*...*/  TSMMD_Count_ };
+	const CSmallMatricesMultiplyDesc* initSmallMatricesMulDescs( TSMMD type, int batch,
+		int firstHeight, int firstWidth, int secondWidth, int resultWidth );
+	void destroySmallMatricesMulDescs( int inputCount );
+
+	int numberOfElements = 0; // the number of elements (neurons) of the fully-connected layer
```

Comment:
_codestyle_ order of declarations/fields/methods

Reply:
fixed